addons/mrp/models/mrp_production.py

```diff
@@ -227,6 +228,16 @@ def _compute_confirm_cancel(self):
         for production in self:
             production.confirm_cancel = productions_with_done_move.get(production.id, False)
 
+    @api.depends('move_raw_ids', 'move_raw_ids.reserved_availability')
+    def _compute_has_ready_produce(self):
+        for order in self:
+            if order.state not in ['draft', 'done', 'cancel']:
+                order.has_ready_produce = all(
+                    move.unit_factor <= move.reserved_availability
+                    for move in order.move_raw_ids)
+            else:
+                order.has_ready_produce = False
```

Comment:
shoudln't we handle the "ready_to_produce" field on the bom here?
``` python
    ready_to_produce = fields.Selection([
        ('all_available', ' When all components are available'),
        ('asap', 'When components for 1st operation are available')], string='Manufacturing Readiness',
```

Reply:
@jip-odoo

Reply:
can we set value for 'has_ready_produce' field based on value of 'ready_to_produce' field on the bom ?
if order.bom_id.ready_to_produce == 'asap' and order.state not in ['draft', 'done', 'cancel']:
         order.has_ready_produce = all(
                    move.unit_factor <= move.reserved_availability
                    for move in order.move_raw_ids)
else:
          order.has_ready_produce = False

Reply:
after discussing it with the PO, the ready to produce field is only available when the wokorders are installed
we suggest to remove this computed field and to make the produce button green once the MO is reserved (use the reservation_state field)

thanks

Reply:
@sle-odoo  here, make produce button green when components are available to produce at least one product.
and here, reservation_state field has only three state confirmed, assigned and waiting.
so here assigned is for all components are available and confirmed is set when components are not available to produce one product.
so which state use in this case?
can we add one another state?
thank you

Reply:
> make the produce button green once the MO is reserved

so juste make the button green if reservation_state == 'assigned', no need for a compute

Reply:
that is already done in this : https://github.com/odoo/odoo/commit/c407d9afb94003898cb24041da3cbbf7311a40de#diff-142aaef733d5def20f10c83875b7bc6bR43
then what about point 7 in spec. ?

Reply:
then the point 7 is already done, we're good :)